src/main/java/org/truffleruby/parser/ast/DStrParseNode.java

```diff
@@ -39,16 +39,26 @@
 
 /** A string which contains some dynamic elements which needs to be evaluated (introduced by #). */
 public class DStrParseNode extends DParseNode implements ILiteralNode {
+    private boolean frozen;
 
-    public DStrParseNode(SourceIndexLength position, Encoding encoding) {
+    public DStrParseNode(SourceIndexLength position, Encoding encoding, boolean frozen) {
         super(position, encoding);
+        setFrozen(frozen);
     }
 
     @Override
     public NodeType getNodeType() {
         return NodeType.DSTRNODE;
     }
 
+    public boolean isFrozen() {
+        return frozen;
+    }
+
+    public void setFrozen(boolean frozen) {
+        this.frozen = frozen;
+    }
+
```

Comment:
Same methods exist on StrParseNode - should I somehow extract this?

Reply:
I wouldn't worry about it - this code we're working is already very badly factored and it needs a big spring-clean really.